src/providers/postgres/raster/qgspostgresrasterprovider.cpp

```diff
@@ -1909,6 +1919,23 @@ QStringList QgsPostgresRasterProvider::parseUriKey( const QString &key )
   return cols;
 }
 
+bool QgsPostgresRasterProvider::checkExtent()
+{
+  // Last check for extent and crs (see GH #39779 crash)
+  if ( ! mCrs.bounds().isEmpty() )
+  {
+    const QgsCoordinateTransform transformer { mCrs, QgsCoordinateReferenceSystem( 4326 ), transformContext() };
+    const QgsRectangle transformedExtent { transformer.transform( mExtent ) };
+    if ( ! mCrs.bounds().contains( transformedExtent ) )
```

Comment:
The cause of the crash is in GDAL because https://github.com/qgis/QGIS/blob/master/src/core/qgsgdalutils.cpp#L163 returns 0x0 and issues a warning:

```
CPLError(CE_Failure, CPLE_AppDefined,
                 "Unable to compute a transformation between pixel/line "
                 "and georeferenced coordinates for %s. "
                 "There is no affine transformation and no GCPs. "
                 "Specify transformation option DST_METHOD=NO_GEOTRANSFORM "
                 "to bypass this check.",
                 GDALGetDescription(hDstDS));
```

And GDAL crashes later when it dereferences the transformer args (GDAL should be probably implement some guards around it but I'm not sure about GDAL's policy in these cases: there is a whole lot of many pointers that should be checked in GDAL's code).

In the  issue test case there was 4326 raster with bound that did not fall into the 4326 bounds, that's probably why GDAL chokes.

I don't know how to handle this error inside the provider `block()` call, because the gdalutils don't throw. But in any event I think it would be better to catch the error on provider construction, if possible, I know `bounds()` is approximated and won't return meaningful results in all cases, but I didn't have a better idea. I'm open to proposals of course.

Reply:
@rouault any ideas here?

Reply:
I've no strong opinion on the fix done here, but I think that QgsGdalUtils::resampleSingleBandRaster() should be modified to return a boolean and GDALCreateGenImgProjTransformer() return should definitely be tested.  A null transformer is an invalid condition for most things that accept a GDALTransformer. The return of ChunkAndWarpImage() should be tested against CE_None as well